src/model/Enemy.hs

```diff
@@ -1,10 +1,131 @@
+    hit:: a -> Bullet -> Maybe (a, Bullet) 
+    -- Render enemy at a random vertical position.
+    spawn :: a
+    -- Return the position of an Enemy
+    pos :: a -> Position
+
+-- Default values, change these later.
+screenWidth :: Int
+screenWidth = 500
+randomY :: Int
+randomY = 300   
+
+-- Implementation of method 'hit' for each given enemy. 
+enemyIsHit :: Enemy e => e -> Bullet -> Maybe (e, Bullet)
+enemyIsHit e b | pos e == bulletPos b = Just (e, b) -- Enemy is hit. Change to this later: if the position is approximately equal.
+               | otherwise = Nothing -- Enemy is not hit.
+
+-- Enemy types as instances of type class Enemy (Too much repitition?)
+instance Enemy BasicEnemy where
+        hit = enemyIsHit
+        spawn = BasicEnemy {basicEnemyPos = (screenWidth, randomY) , basicEnemyCooldown = cooldown} -- Spawn an enemy at right with a random y pos.
+                where -- Default value, change this later.
+                        cooldown = 2
+        pos = basicEnemyPos
+
+instance Enemy BurstEnemy where
+        hit = enemyIsHit
+        spawn = BurstEnemy {burstEnemyPos = (screenWidth, randomY) , burstEnemyCooldown = cooldown}  
+                where
+                        cooldown = 3
+        pos = burstEnemyPos
+
+instance Enemy ConeEnemy where
+        hit = enemyIsHit
+        spawn = ConeEnemy {coneEnemyPos = (screenWidth, randomY) , coneEnemyCooldown = cooldown}    
+                where
+                        cooldown = 2
+        pos = coneEnemyPos
+
+instance Enemy BasicPlayerSeekingEnemy where
+        hit = enemyIsHit
+        spawn = BasicPlayerSeekingEnemy {basicSeekingPos = (screenWidth, randomY) , basicSeekingCooldown = cooldown}      
+                where
+                        cooldown = 1
+        pos = basicSeekingPos
+
+instance Enemy FastPlayerSeekingEnemy where
+        hit = enemyIsHit
+        spawn = FastPlayerSeekingEnemy {fastSeekingPos = (screenWidth, randomY)}     
+        pos = fastSeekingPos
+       
+
+-- All enemies, except the player seeking types, are instances of this class.
+class MoveBasic a where 
+        moveBasic :: a -> a 
+    
+instance MoveBasic BasicEnemy where
+        moveBasic b = let (x, y) = pos b in
+                b {basicEnemyPos = (x - 1, y)}
+
+instance MoveBasic BurstEnemy where
+        moveBasic b = let (x, y) = pos b in
+                b {burstEnemyPos = (x - 1, y)}
+
+instance MoveBasic ConeEnemy where
+        moveBasic c = let (x, y) = pos c in
+                c {coneEnemyPos = (x - 1, y)} 
+
+
+    -- Only the two player seeking enemies are instances of this class.
+class MovePlayerSeeking a where 
+        moveSeeking :: a -> Player -> a        
+
+instance MovePlayerSeeking BasicPlayerSeekingEnemy where
+        moveSeeking b player = let (x, y) = pos b in
+                b {basicSeekingPos = (x - 1, y + seeking_speed * abs (y - playerPos player))} -- The vertical movement is based on the vertical distance between player and enemy.
```

Comment:
Ik denk dat je de `abs` niet moet hebben hier, want nu kan hij alleen naar boven bewegen toch? Als het verschil negatief is moet hij ook naar beneden kunnen.